aggregator_core/src/datastore/models.rs

```diff
@@ -774,41 +838,54 @@ where
 impl<const SEED_SIZE: usize, A: vdaf::Aggregator<SEED_SIZE, 16>> Eq
     for ReportAggregation<SEED_SIZE, A>
 where
-    A::PrepareState: Eq,
-    A::PrepareMessage: Eq,
+    A::InputShare: Eq,
     A::PrepareShare: Eq,
+    A::PublicShare: Eq,
     A::OutputShare: Eq,
 {
 }
 
 /// ReportAggregationState represents the state of a single report aggregation. It corresponds
 /// to the REPORT_AGGREGATION_STATE enum in the schema, along with the state-specific data.
-#[derive(Clone, Debug, Derivative)]
+#[derive(Clone, Derivative)]
+#[derivative(Debug)]
 pub enum ReportAggregationState<const SEED_SIZE: usize, A: vdaf::Aggregator<SEED_SIZE, 16>> {
-    Start,
-    WaitingLeader(
+    StartLeader {
+        /// Public share for this report.
+        public_share: A::PublicShare,
+        /// The sequence of extensions from the Leader's input share for this report.
+        leader_extensions: Vec<Extension>,
```

Comment:
I think we will want to `Debug = "ignore"` these fields as well. It might be nice if the `Debug` impl on `Extension` printed the extension type but not the contents, but that's not needed in this change.